node/src/components/block_proposer.rs

```diff
@@ -391,21 +409,24 @@ impl BlockProposerReady {
                 break;
             }
 
-            if self.is_deploy_valid(
-                deploy_type.header(),
-                block_timestamp,
-                &deploy_config,
-                &past_deploys,
-            ) && !past_deploys.contains(hash)
+            if self.is_deploy_valid(&deploy_type, block_timestamp, &deploy_config, &past_deploys)
+                && !past_deploys.contains(hash)
                 && !self.sets.finalized_deploys.contains_key(hash)
             {
+                let header = deploy_type.header();
+                let gas_price = header.gas_price();
+                if gas_price + block_gas_running_total > deploy_config.block_gas_limit {
+                    break;
```

Comment:
This might just be a particularly big deploy — the next one might fit into the block again.
We'll need to sort the deploys:
* Highest gas _price_ comes first.
* For equal gas price, highest _gas limit_ comes first.

I believe we also want to allow transfers in the general section of the block: the one for the Wasm deploys.
So: The transfer part is reserved for transfers only, but transfers that outbid all remaining Wasm deploys should still get into the block, even outside of the transfer part.

(@AlexanderLimonov: Please correct me if I'm wrong!)

Reply:
Yes, that's the correct ordering, the idea is to maximize revenue and to enable a coherent mapping from these deploy orderings to a "price," which, with reasonably small deploys dominating, would just be the marginal willingness to pay in the last block.

It seems I may not have written the provision for allowing overflowing WASMless transfers into the general section, but I believe that should be done (letting these transfers compete on price and volume as usual with regular deploys).

I think the final version of the ordering is this - lump *all* deploys together and sort (first by price, then by reservation), then go down the list and include each in their respective sections until WASMless transfers overflow, proceeding as normal until the regular deploy gas limit is reached.

Reply:
Per jira comment, we'll be addressing this after CEP22 merges.